sphinx/registry.py

```diff
@@ -239,10 +239,23 @@ def add_source_suffix(self, suffix, filetype):
         else:
             self.source_suffix[suffix] = filetype
 
-    def add_source_parser(self, suffix, parser):
-        # type: (unicode, Type[Parser]) -> None
-        logger.debug('[app] adding search source_parser: %r, %r', suffix, parser)
-        self.add_source_suffix(suffix, suffix)
+    def add_source_parser(self, *args):
```

Comment:
nit: How about doing this instead?

    empty_parser = object()

    def add_source_parser(self, suffix, parser=empty_parser):
        if parser != empty_parser:
            # emit warning here

It would keep a similarly (clear) function style and allow you to do the same thing. It works because every `object` instance is unique. You could also use `parser=None`, but if the user called this with `None` intentionally, they wouldn't see the issue

Reply:
Ohhh, wait a second. I mixed up the order of the parameters. `suffix` is the first parameter - not the second. You can ignore this comment so 😄

Reply:
Okay :-)